appserver/payara-appserver-modules/microprofile/openapi/src/main/java/fish/payara/microprofile/openapi/impl/OpenAPISupplier.java

```diff
@@ -0,0 +1,264 @@
+import org.glassfish.hk2.classmodel.reflect.Type;
+import org.glassfish.hk2.classmodel.reflect.Types;
+import org.glassfish.internal.api.Globals;
+import org.glassfish.internal.api.ServerContext;
+import org.glassfish.internal.deployment.analysis.StructuredDeploymentTracing;
+
+import fish.payara.microprofile.openapi.impl.config.OpenApiConfiguration;
+import fish.payara.microprofile.openapi.impl.model.OpenAPIImpl;
+import fish.payara.microprofile.openapi.impl.processor.ApplicationProcessor;
+import fish.payara.microprofile.openapi.impl.processor.BaseProcessor;
+import fish.payara.microprofile.openapi.impl.processor.FileProcessor;
+import fish.payara.microprofile.openapi.impl.processor.FilterProcessor;
+import fish.payara.microprofile.openapi.impl.processor.ModelReaderProcessor;
+
+public class OpenAPISupplier implements Supplier<OpenAPI> {
+
+    private final OpenApiConfiguration config;
+    private final String applicationId;
+    private final String contextRoot;
+    private final ReadableArchive archive;
+    private final ClassLoader classLoader;
+
+    private volatile OpenAPI document;
+
+    private boolean enabled;
+
+    public OpenAPISupplier(String applicationId, String contextRoot,
+            ReadableArchive archive, ClassLoader classLoader) {
+        this.config = new OpenApiConfiguration(classLoader);
+        this.applicationId = applicationId;
+        this.contextRoot = contextRoot;
+        this.archive = archive;
+        this.classLoader = classLoader;
+        this.enabled = true;
+    }
+
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    @Override
+    public synchronized OpenAPI get() {
+        if (this.document != null) {
+            return this.document;
+        }
+        if (!enabled) {
+            return null;
+        }
+
+        try {
+            final Parser parser = Globals.get(ApplicationLifecycle.class).getDeployableParser(
+                    archive,
+                    true,
+                    true,
+                    StructuredDeploymentTracing.create(applicationId),
+                    Logger.getLogger(OpenApiService.class.getName())
+            );
+            final Types types = parser.getContext().getTypes();
+
+            OpenAPI doc = new OpenAPIImpl();
+            try {
+                final List<URL> baseURLs = getServerURL(contextRoot);
+                doc = new ModelReaderProcessor().process(doc, config);
+                doc = new FileProcessor(classLoader).process(doc, config);
+                doc = new ApplicationProcessor(
+                        types,
+                        filterTypes(archive, config, types),
+                        classLoader
+                ).process(doc, config);
+                doc = new BaseProcessor(baseURLs).process(doc, config);
+                doc = new FilterProcessor().process(doc, config);
+            } finally {
+                this.document = doc;
```

Comment:
Does this work like as intended should an exception be thrown at some of the steps so that the document contains as much information as possible to that point?

Reply:
My changes don't (see "shouldn't") change any functionality when the sniffer returns `true` for the application. All DeploymentExceptions related to MP deployment should still propagate, this just disables all CDI extensions and processing when no MP or JAX-RS annotations are detected